openmetadata-service/src/main/java/org/openmetadata/service/security/saml/SamlAssertionConsumerServlet.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.service.security.saml;
+
+import com.onelogin.saml2.Auth;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.auth.JWTAuthMechanism;
+import org.openmetadata.schema.auth.ServiceTokenType;
+import org.openmetadata.service.security.jwt.JWTTokenGenerator;
+
+@WebServlet("/api/v1/saml/acs")
+@Slf4j
+public class SamlAssertionConsumerServlet extends HttpServlet {
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) {
+    try {
+      handleResponse(req, resp);
+    } catch (Exception e) {
+      LOG.error("SamlResponseError :" + e.getMessage());
+    }
+  }
+
+  private void handleResponse(HttpServletRequest req, HttpServletResponse resp) throws Exception {
+    Auth auth = new Auth(SamlSettingsHolder.getInstance().getSaml2Settings(), req, resp);
+    auth.processResponse();
+    if (!auth.isAuthenticated()) {
+      LOG.error("[SAML ACS] Not Authenticated");
+      resp.sendError(403, "Unauthorized");
```

Comment:
`Not Authenticated` or `Unauthorized`? We are logging and sending different things.

Reply:
fixed